tests/unit/utils/test_zarr_methods.py

```diff
@@ -1337,3 +1337,347 @@ def test_update_zarr(manager_class):
+        dm.to_zarr = mock.Mock()
+
+        update_dataset = object()
+        insert_times = object()
+        original_chunks = object()
+
+        append_dataset = dm.prep_update_dataset.return_value = mock.MagicMock()
+        append_dataset.attrs = {}
+
+        mapper = dm.store.mapper.return_value
+        mapper.freeze.return_value = 42
+
+        dm.append_to_dataset(update_dataset, insert_times, original_chunks)
+
+        dm.store.mapper.assert_called_once_with()
+        dm.prep_update_dataset.assert_called_once_with(update_dataset, insert_times, original_chunks)
+
+        dm.to_zarr.assert_called_once_with(append_dataset, mapper, consolidated=True, append_dim="time")
+
+        assert append_dataset.attrs == {"update_is_append_only": True}
+        assert dm.dataset_hash == "42"
+
+    @staticmethod
+    def test_append_to_dataset_not_ipld_dry_run(manager_class):
+        dm = manager_class()
+        dm.dataset_hash = "browns"
+        dm.dry_run = True
+        dm.store = mock.Mock(spec=store.StoreInterface)
+        dm.prep_update_dataset = mock.Mock()
+        dm.calculate_update_time_ranges = mock.Mock()
+        dm.to_zarr = mock.Mock()
+
+        update_dataset = object()
+        insert_times = object()
+        original_chunks = object()
+
+        append_dataset = dm.prep_update_dataset.return_value = mock.MagicMock()
+        append_dataset.attrs = {}
+
+        mapper = dm.store.mapper.return_value
+        mapper.freeze.return_value = 42
+
+        dm.append_to_dataset(update_dataset, insert_times, original_chunks)
+
+        dm.store.mapper.assert_called_once_with()
+        dm.prep_update_dataset.assert_called_once_with(update_dataset, insert_times, original_chunks)
+
+        dm.to_zarr.assert_called_once_with(append_dataset, mapper, consolidated=True, append_dim="time")
+
+        assert append_dataset.attrs == {"update_is_append_only": True}
+        assert dm.dataset_hash == "browns"
+
+    @staticmethod
+    def test_prep_update_dataset(manager_class, fake_complex_update_dataset):
+        # Give the transpose call in prep_update_dataset something to do
+        dataset = fake_complex_update_dataset.transpose("longitude", "latitude", "time")
+        chunks = {"time": 5}
+        time_values = numpy.arange(
+            numpy.datetime64("2022-02-01T00:00:00.000000000"),
+            numpy.datetime64("2022-03-09T00:00:00.000000000"),
+            numpy.timedelta64(1, "[D]"),
+        )
+        dm = manager_class()
+        dm.set_zarr_metadata = lambda x: x
+
+        assert len(dataset.time) > len(time_values)
+
+        dataset = dm.prep_update_dataset(dataset, time_values, chunks)
+
+        assert numpy.array_equal(dataset.time, time_values)
+        dataset.chunks["time"][0] == 5
+        dataset.chunks["latitude"][0] == 4
+        dataset.chunks["longitude"][0] == 4
+
+        # TODO: I'd like to be able to verify order of coordinates after transpose, but can't figure out how get
```

Comment:
Perhaps you could look at the order of chunks / coordinates for the data variable array? This is what really matters, after all, as coordinate order there affects writing and retrieval speeds.